c7n/resources/org.py

```diff
@@ -191,8 +191,133 @@ def parse_query(self):
+@OrgUnit.filter_registry.register("policy")
+@OrgAccount.filter_registry.register("policy")
+class PolicyFilter(ListItemFilter):
+    schema = type_schema(
+        "policy",
+        required=["policy-type"],
+        **{
+            "policy-type": {"enum": OrgPolicy.policy_types},
+            "inherited": {"type": "boolean"},
+            "attrs": {"$ref": "#/definitions/filters_common/list_item_attrs"},
+            "count": {"type": "number"},
+            "count_op": {"$ref": "#/definitions/filters_common/comparison_operators"},
+        },
+    )
+
+    permissions = ("organizations:ListRoots", "organizations:ListPoliciesForTarget")
+
+    annotate_items = True
+    annotation_key = "c7n:PolicyMatches"
+    target_policies = None
+    ou_root = None
+    client = None
+
+    def process(self, resources, event):
+        self.client = local_session(self.manager.session_factory).client("organizations")
+        if self.data.get("inherited") and self.manager.type == "org-account":
+            # Get ou account hierarchy / parents
+            hierarchy_manager = self.manager.get_resource_manager(
+                "org-account", {"filters": ["org-unit"]}
+            )
+
+            ou_assembly = hierarchy_manager.filters[0]
+            ou_assembly.ou_map = {
+                ou["Id"]: ou for ou in self.manager.get_resource_manager("org-unit").resources()
+            }
+            ou_assembly.process_accounts(resources, event)
+            # also initialize root for accounts as we dont store it as a parent.
+            self.ou_root = self.client.list_roots()["Roots"][0]
+        self.target_policies = {}
+        return super().process(resources, event)
+
+    def get_targets(self, resource):
+        if not self.data.get("inherited"):
+            yield resource["Id"]
+            return
+
+        # return self, parents, root
+
+        # handle ous
+        if self.manager.type == "org-unit":
+            yield resource["Id"]
+            for p in resource["Parents"]:
+                yield p
+            return
+
+        # handle accounts
+        yield resource["Id"]
+        for p in resource[OrgUnitFilter.annotation_parent_key]:
+            yield p["Id"]
+        yield self.ou_root["Id"]
+
+    def get_item_values(self, resource):
+        rpolicies = {}
+        for tgt_id in self.get_targets(resource):
+            if tgt_id not in self.target_policies:
+                policies = self.client.list_policies_for_target(
+                    Filter=self.data["policy-type"], TargetId=tgt_id
+                ).get("Policies", ())
+                self.target_policies[tgt_id] = policies
+            for p in self.target_policies[tgt_id]:
+                rpolicies[p["Id"]] = p
+        return list(rpolicies.values())
+
+
+@OrgAccount.action_registry.register("set-policy")
+@OrgUnit.action_registry.register("set-policy")
+class SetPolicy(Action):
+    """Set a policy on an org unit or account"""
```

Comment:
I was testing some scenarios locally using examples SCPs (since folks may look at AWS docs examples and see how to implement them in c7n). Maybe this would be a useful docstring addition:

```suggestion
    """Set a policy on an org unit or account

    .. code-block:: yaml

        policies:
          - name: attach-existing-scp
            resource: aws.org-unit
            filters:
              - type: policy
                policy-type: SERVICE_CONTROL_POLICY
                count: 0
                attrs:
                  - Name: RestrictedRootAccount
            actions:
              - type: set-policy
                policy-type: SERVICE_CONTROL_POLICY
                name: RestrictedRootAccount

    .. code-block:: yaml

        policies:
          - name: create-and-attach-scp
            resource: aws.org-unit
            filters:
              - type: policy
                policy-type: SERVICE_CONTROL_POLICY
                count: 0
                attrs:
                  - Name: RestrictedRootAccount
            actions:
              - type: set-policy
                policy-type: SERVICE_CONTROL_POLICY
                name: RestrictedRootAccount
                contents:
                  Version: "2012-10-17"
                  Statement:
                    - Sid: RestrictEC2ForRoot
                      Effect: Deny
                      Action:
                        - "ec2:*"
                      Resource:
                        - "*"
                      Condition:
                        StringLike:
                          "aws:PrincipalArn":
                            - arn:aws:iam::*:root
    """
```